py-rattler/rattler/repo_data/package_record.py

```diff
@@ -119,6 +119,37 @@ def to_graph(records: List[PackageRecord]) -> nx.DiGraph:  # type: ignore[type-a
 
         return graph
 
+    @staticmethod
+    def validate(records: List[PackageRecord]) -> None:
+        """
+        Validate that the given package records are valid w.r.t. 'depends' and 'constrains'.
+
+        This function will return nothing if all records form a valid environment, i.e., all dependencies
+        of each package are satisfied by the other packages in the list.
+        If there is a dependency that is not satisfied, this function will raise an exception.
+
+        Examples
+        --------
+        ```python
+        >>> from os import listdir
+        >>> from os.path import isfile, join
+        >>> from rattler import PrefixRecord
+        >>> from rattler.exceptions import ValidatePackageRecordsException
+        >>> records = [
+        ...     PrefixRecord.from_path(join("../test-data/conda-meta/", f))
+        ...     for f in sorted(listdir("../test-data/conda-meta"))
```

Comment:
The sorted here is only to make this reproducible. If you have better ideas I'm open for them